clifford/test/test_dpga.py

```diff
@@ -0,0 +1,188 @@
+
+import numpy as np
+import pytest
+from clifford.dpga import *
+
+
+class TestBasicDPGA:
+    def test_non_orthogonal_metric(self):
+        for i, a in enumerate(wbasis):
+            for j, b in enumerate(wbasis):
+                res = a|b
+                if i != j and i%4 == j%4:
+                    assert res == (0.5) + 0*w1
+                else:
+                    assert res == 0*w1
+                print(a|b, end=' ')
+            print()
+
+    def test_bivector_identities(self):
+        for wi in wlist:
+            for wj in wlist:
+                assert wi^wj == -wj*wi
+
+        for wis in wslist:
+            for wjs in wslist:
+                assert wis^wjs == -wjs*wis
+
+        for w in wbasis:
+            assert w**2 == 0*w1
+
+        for wi, wis in zip(wlist, wslist):
+            assert wi*wis == 1 - wis*wi
+
+    def test_up_down(self):
+        for i in range(1000):
+            p = np.random.randn(3)
+            dpga_pnt = up(p)
+            pnt_down = down(np.random.rand()*dpga_pnt)
+            np.testing.assert_allclose(pnt_down, p)
+
+    def test_translate(self):
+        for i in range(100):
+            tvec = np.random.randn(3)
```

Comment:
To make seeding a little easier and threadsafe,
```suggestion
        rng = np.random.default_rng()   # can pass a seed here later
        for i in range(100):
            tvec = rng.standard_normal()
```
same throughout